### PR TITLE
fix that PMacc depends on PIConGPU type definitions

### PR DESCRIPTION
**Pull request by psychocoderHPC**
PMacc should be independent of any PIConGPU types. Fix using PIConGPU
types.

### CODE REVIEW

File: include/pmacc/fields/tasks/TaskFieldReceiveAndInsert.hpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ namespace pmacc
     class TaskFieldReceiveAndInsert : public MPITask
     {
     public:
-        static constexpr uint32_t Dim = picongpu::simDim;
+        static constexpr uint32_t Dim = Field::dim;
```
Comment (sbastrakov) [inline include/pmacc/fields/tasks/TaskFieldReceiveAndInsert.hpp:39]: Not really important, but should `#include <cstdint>`. Same in the next file

File: include/pmacc/math/complex/Complex.tpp [hunk 2]
```diff
@@ -277,12 +278,21 @@ namespace pmacc
 
     namespace mpi
     {
-        using complex_X = pmacc::math::Complex<picongpu::float_X>;
+        // Specialize complex type grid buffer for MPI
+        template<>
+        HINLINE MPI_StructAsArray getMPI_StructAsArray<pmacc::math::Complex<float>>()
+        {
+            using complex_X = pmacc::math::Complex<float>;
```
Comment (sbastrakov) [inline include/pmacc/math/complex/Complex.tpp:285]: Maybe this type should be called `Complex` ? Same later

**Review by sbastrakov — APPROVED** [review]
Some small suggestions above, but not important. Can be merged as is

File: include/pmacc/math/complex/Complex.tpp (hunk 2 above)
Comment (psychocoderHPC) [reply include/pmacc/math/complex/Complex.tpp:285]: I will change it because I need to touch this PR because of the other suggestion anyway.

**Review by sbastrakov — APPROVED** [review]

**Pull request merged**